### PR TITLE
catch environment failures in run policy

### LINKED ISSUE #835: handle case where an environment failure is raised during `run_policy` in `_generate_interaction_results`
an annoyance is that we can't just add `utils.EnvironmentFailure` to `exceptions_to_break_on` because those exceptions are only caught when they get raised in `policy(state)`, not when they're in `env.step(act)`

### PR DESCRIPTION
**Pull request by tomsilver**
closes #835

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -839,7 +839,11 @@ def run_policy(
     Terminates when any of these conditions hold:
     (1) the termination_function returns True
     (2) max_num_steps is reached
-    (3) policy() raises any exception of type in exceptions_to_break_on
+    (3) policy() or step() raise an exception of type in exceptions_to_break_on
+
+    Note that in the case where the exception is raised in step, we exclude the
+    last action from the returned trajectory to maintain the invariant that
+    the trajectory states are always one more than the actions.
```
Comment (ronuchit) [inline src/utils.py:846]: are of length one greater than the actions*

File: src/utils.py [hunk 2]
```diff
@@ -895,7 +899,11 @@ def run_policy_with_simulator(
...
```
Comment (ronuchit) [inline src/utils.py:906]: copy-paste change above

File: src/utils.py [hunk 3]
```diff
@@ -839,7 +839,11 @@ def run_policy(
     Terminates when any of these conditions hold:
     (1) the termination_function returns True
     (2) max_num_steps is reached
-    (3) policy() raises any exception of type in exceptions_to_break_on
+    (3) policy() or step() raise an exception of type in exceptions_to_break_on
```
Comment (ronuchit) [inline src/utils.py:842]: raises*

File: src/utils.py [hunk 4]
```diff
@@ -895,7 +899,11 @@ def run_policy_with_simulator(
...
```
Comment (ronuchit) [inline src/utils.py:902]: raises*

**Review by ronuchit — APPROVED** [review]

File: src/utils.py (hunk 3 above)
Comment (tomsilver) [reply src/utils.py:842]: You want this change even though the line will break?

Comment (ronuchit) [reply src/utils.py:842]: no

**Pull request merged**